plugins/rsyslog.yaml

```
@@ -0,0 +1,105 @@
+    default: "UTC"
+    relevant_if:
+      protocol:
+        equals: "rfc3164 (BSD)"
+
+# Set Defaults
+# {{$listen_address := default "0.0.0.0:514" .listen_address}}
+# {{$connection_type := default "udp" .connection_type}}
+# {{$protocol := default "rfc5424 (IETF)" .protocol}}
+# {{$location := default "UTC" .location}}
+
+# Pipeline Template
+pipeline:
+# {{ if eq $connection_type "udp" }}
+  - id: rsyslog_input
+    type: udp_input
+    listen_address: {{ $listen_address }}
+    labels:
+      log_type: rsyslog
+      plugin_id: {{ .id }}
+    output: rsyslog_parser
+# {{ end }}
+
+# {{ if eq $connection_type "tcp" }}
+  - id: rsyslog_input
+    type: tcp_input
+    listen_address: {{ $listen_address }}
+    labels:
+      log_type: rsyslog
+      plugin_id: {{ .id }}
+    output: handle_new_lines
+
+  # tcp input uses \n to break messages into seperate
+  # entries, this can be a problem for Syslog if the
+  # structured_data key=value pairs or message fields contain
+  # a \n. This is unlikly, but not impossible. It has been
+  # observed with rfc5424 structured_data field.
+  # '<' can be relied on to be the first character of every
+  # syslog message for both rfc 5424 and 3164
+  - id: handle_new_lines
+    type: recombine
+    combine_field: $record
+    is_first_entry: "$record startsWith '<'"
+    output: return_router
```

The original Syslog implementation needs to handle special situations where syslog entries are multi line. This should not be the case for a dedicated rsyslog plugin. 

```suggestion
  - id: rsyslog_input
    type: tcp_input
    listen_address: {{ $listen_address }}
    labels:
      log_type: rsyslog
      plugin_id: {{ .id }}
    output: return_router
```

---

rsyslog_input will need to output to `rsyslog_parser`, see next comment below V V V